MoreLinq.Test/NullArgumentTest.cs

```diff
@@ -142,7 +142,11 @@ static object CreateInstance(Type type)
             if (type.GetTypeInfo().IsValueType || HasDefaultConstructor(type)) return Activator.CreateInstance(type);
             if (typeof(Delegate).IsAssignableFrom(type)) return CreateDelegateInstance(type);
 
-            return CreateGenericInterfaceInstance(type.GetTypeInfo());
+            var typeInfo = type.GetTypeInfo();
+
+            return typeInfo.IsGenericType
+                    ? CreateGenericInterfaceInstance(typeInfo)
+                    : CreateNonGenericInterfaceInstance(typeInfo);
```

Comment:
I don't understand the purpose of `CreateNonGenericInterfaceInstance`. Why not just use `new NonGenericArgs.Enumerable()`?

Reply:
I tried to follow the pattern, but really it is not necessary here